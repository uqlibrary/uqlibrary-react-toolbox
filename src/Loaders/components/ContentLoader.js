import React from 'react';
import {PropTypes} from 'prop-types';
import {Card, CardText} from 'material-ui/Card';
import InlineLoader from './InlineLoader';

export default function ContentLoader({message}) {
    return (
        <Card className="layout-card">
            <CardText className="column layout-fill align-center justify-center">
                <InlineLoader message={message} />
            </CardText>
        </Card>
    );
}

ContentLoader.propTypes = {
    message: PropTypes.string
};
