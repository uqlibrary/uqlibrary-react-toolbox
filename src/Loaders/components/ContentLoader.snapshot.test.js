jest.dontMock('./ContentLoader');

import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import ContentLoader from './ContentLoader';

function setup(message) {
    const props = {message};
    return shallow(<ContentLoader {...props} />);
}

describe('ContentLoader snapshots tests', () => {
    it('renders loader', () => {
        const wrapper = setup('Waiting to load...');
        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });
});