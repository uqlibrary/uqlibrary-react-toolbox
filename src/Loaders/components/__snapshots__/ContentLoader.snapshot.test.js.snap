// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentLoader snapshots tests renders loader 1`] = `
<Card
  className="layout-card"
  expandable={false}
  expanded={null}
  initiallyExpanded={false}
>
  <CardText
    className="column layout-fill align-center justify-center"
  >
    <InlineLoader
      message="Waiting to load..."
    />
  </CardText>
</Card>
`;
